egs/wsj/s5/steps/nnet3/report/generate_plots.py

```diff
@@ -279,64 +350,50 @@ def generate_nonlin_stats_plots(exp_dir, output_dir, plot, comparison_dir=None,
             provided only for common component names. Make sure that these are
             comparable experiments before analyzing these plots.""")
-
+        
         fig = plt.figure()
+        
+        common_prefix = os.path.commonprefix(dirs)
+        prefix_length = common_prefix.rfind('/')
+        common_prefix = common_prefix[0:prefix_length]
+        
         for component_name in main_component_names:
-            fig.clf()
-            index = 0
-            plots = []
-            for dir in dirs:
-                color_val = g_plot_colors[index]
-                index += 1
-                try:
-                    iter_stats = (
-                        stat_tables_per_component_per_dir[dir][component_name])
-                except KeyError:
-                    # this component is not available in this network so lets
-                    # not just plot it
-                    continue
-
-                data = np.array(iter_stats)
-                data = data[data[:, 0] >= start_iter, :]
-                ax = plt.subplot(211)
-                mp, = ax.plot(data[:, 0], data[:, 1], color=color_val,
-                              label="Mean {0}".format(dir))
-                msph, = ax.plot(data[:, 0], data[:, 1] + data[:, 2],
-                                color=color_val, linestyle='--',
-                                label="Mean+-Stddev {0}".format(dir))
-                mspl, = ax.plot(data[:, 0], data[:, 1] - data[:, 2],
-                                color=color_val, linestyle='--')
-                plots.append(mp)
-                plots.append(msph)
-                ax.set_ylabel('Value-{0}'.format(comp_type))
-                ax.grid(True)
-
-                ax = plt.subplot(212)
-                mp, = ax.plot(data[:, 0], data[:, 3], color=color_val)
-                msph, = ax.plot(data[:, 0], data[:, 3] + data[:, 4],
-                                color=color_val, linestyle='--')
-                mspl, = ax.plot(data[:, 0], data[:, 3] - data[:, 4],
-                                color=color_val, linestyle='--')
-                ax.set_xlabel('Iteration')
-                ax.set_ylabel('Derivative-{0}'.format(comp_type))
-                ax.grid(True)
-
-            lgd = plt.legend(handles=plots, loc='lower center',
-                             bbox_to_anchor=(0.5, -0.5 + len(dirs) * -0.2),
-                             ncol=1, borderaxespad=0.)
-            plt.grid(True)
-            fig.suptitle("Mean and stddev of the value and derivative at "
-                         "{comp_name}".format(comp_name=component_name))
-            comp_name = latex_compliant_name(component_name)
-            figfile_name = '{dir}/nonlinstats_{comp_name}.pdf'.format(
-                dir=output_dir, comp_name=comp_name)
-            fig.savefig(figfile_name, bbox_extra_artists=(lgd,),
+            if stats_per_dir[exp_dir][component_name]['type'] == 'LstmNonlinearity':
```

Comment:
This code could benefit from few comments. Sorry that I didn't add any even in the original code. Could you please do it in this commit ?